addons/hr_expense/models/hr_expense.py

```diff
@@ -104,6 +104,14 @@ def _default_employee_id(self):
     duplicate_expense_ids = fields.Many2many(comodel_name='hr.expense', compute='_compute_duplicate_expense_ids')  # Used to trigger warnings
     same_receipt_expense_ids = fields.Many2many(comodel_name='hr.expense', compute='_compute_same_receipt_expense_ids')  # Used to trigger warnings
 
+    split_expense_ids = fields.Many2many(
+        comodel_name='hr.expense',
+        relation='hr_expense_split_rel',
+        column1='expense_id',
+        column2='related_expense_id',
+        string="Related Split Expenses",
+        help="All expenses related to this split, including the original and the splits.",
```

Comment:
Isn't it a bit strange, though ? Plus, this will create a LOT of relational links, since every expense of the split will always need to be linked to all the others. Isn't it too much ? Wouldn't a m2o on the split linking it to the main expense be enough ? 

Even if we want to keep this flow as-is, I think it would be possible to implement it like that, and have a bit of python processing to gather the records we want before opening them.